src/main/java/org/commcare/core/network/ModernHttpRequester.java

```diff
@@ -193,12 +136,10 @@ public static void processResponse(HttpResponseProcessor responseProcessor,
             try {
                 responseStream = streamAccessor.getResponseStream();
             } catch (IOException e) {
-                responseProcessor.handleIOException(e);
+                responseProcessor.handleException(e);
```

Comment:
Is there a specific reason why we eliminated the specific exception typing on the response processor?

I think in the abstract sense, blurring an IOException to an untyped Exception seems like a bad thing.

Reply:
Agree. Though it was done to incorporate `PlainTextPasswordException`. If we make it an `IOException` then I think we would be able to retain the earlier exception typing

Reply:
One other option would actually be to make the response processor have two different methods. 

Generally speaking in this context I'd expect an I/O failure to have a different semantic meaning than a generic exception, so there's some argument to be made that it makes sense for there to be different error handling paths. 

I think either of the two options laid out (Making PlainText an I/O Exception, or making there be two return paths in the error handling) feel like they should work to me, open to a third option as well.